### PR TITLE
Added configuration for view precompilation.

### PR DESCRIPTION
**Pull request by MichalTichy**
This PR adds configuration for view compilation modes.
```
/// Gets or sets the mode under which the view compilation (pages, controls, ... ) is done.
configuration.Markup.ViewCompilation.Mode = ViewCompilationMode.AfterStartup;

/// Gets or sets the delay before view compilation will be done. This compilation delay can be set only in precompilation modes.
configuration.Markup.ViewCompilation.BackgroundCompilationDelay = TimeSpan.FromMinutes(1);
```

Compilation modes are:
- Lazy 
Current mode of compilation.
- DuringApplicationStart
Compilation will run during application startup. Application will start after compilation is done.
- AfterApplicationStart
Compilation will run after application startup.

### CODE REVIEW

File: src/DotVVM.Framework/Configuration/ViewCompilationConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                ThrowIfFrozen();
+                compileInParallel = value;
+            }
+        }
+
+        public void Validate()
+        {
+            if (BackgroundCompilationDelay.HasValue && Mode == ViewCompilationMode.Lazy)
+            {
+                throw new Exception($"{nameof(BackgroundCompilationDelay)} is not supported in {nameof(ViewCompilationMode.Lazy)} {nameof(Mode)}.");
```
Comment (exyi) [inline src/DotVVM.Framework/Configuration/ViewCompilationConfiguration.cs:57]: BackgroundCompilationDelay must be null in ...

File: src/DotVVM.Framework/Controls/UpdateProgress.cs [hunk 2]
```diff
@@ -93,7 +93,7 @@ public static IEnumerable<ControlUsageError> ValidateUsage(ResolvedControl contr
             {
                 if ((int)delayProperty.Value < 0)
                 {
-                    yield return new ControlUsageError("Delay cannot be set to negative number.");
+                    yield return new ControlUsageError("BackgroundCompilationDelay cannot be set to negative number.");
```
Comment (exyi) [inline src/DotVVM.Framework/Controls/UpdateProgress.cs:96]: No, this is wrong :sweat_smile:

File: src/DotVVM.Framework/Configuration/ViewCompilationConfiguration.cs (hunk 1 above)
Comment (exyi) [reply src/DotVVM.Framework/Configuration/ViewCompilationConfiguration.cs:57]: We should also check that the delay is null when Mode is DuringStartup. That would only stop the startup, which does not make much sense.

Comment (MichalTichy) [reply src/DotVVM.Framework/Configuration/ViewCompilationConfiguration.cs]: Fixed

File: src/DotVVM.Framework/Controls/UpdateProgress.cs (hunk 2 above)
Comment (MichalTichy) [reply src/DotVVM.Framework/Controls/UpdateProgress.cs]: Ups :O
I've reverted the change and modified all the messages in UpdateProgress to use nameof

**Comment by tomasherceg** [comment]
It seems like that integration tests are failing - @MichalTichy please look at it.

**Comment by MichalTichy** [comment]
Tests seem to be ok now.
Can we merge it?
@tomasherceg

**Pull request merged**